### DISCUSSION TITLE
Access Policy

### DISCUSSION THREAD

**Discussion by majva-g** [fourdotsix/tenant-buckets#24] [General]
created bucket 'Access Policy' is private.
How to create a public bucket?
adding  'visibility' => 'public', to the filesystem config doesn't affect.

**Comment by secrethash** [comment]
Hey @majva-g ,
Can you answer a few questions before I can help you out:

- Is the question related to Minio or S3 or any other S3 compatible service?
- Are you not able to create a bucket without an access policy as public?

**Reply by majva-g** [reply]
I am using minio.
Everything is ok. The bucket is created when the tenant is created. 
My only problem is that after creation I have to manually log in to minio and make the Access Policy 
![Screenshot from 2024-12-11 17-18-23](https://github.com/user-attachments/assets/a591b0b1-ed71-49d8-b366-b6ea63f23021)

**Reply by majva-g** [reply]
Are you not able to create a bucket without an access policy as public?
No, without an access policy  it will be private

**Reply by secrethash** [reply]
Oh I get it. I am actually working on a feature to add policies during creation. But for now if you want to do it manually you can add a listener to attach the policies using AWS S3 SDK for the created bucket, after it has been created. You will need to listen and map your listener to this event: https://github.com/vidwanco/tenant-buckets/blob/tenancy-v4/src/Events/CreatedBucket.php
